client/acl.go

```diff
@@ -81,7 +81,7 @@ func (c *Client) ResolveSecretToken(secretID string) (*structs.ACLToken, error)
 
 func (c *Client) resolveTokenAndACL(secretID string) (*acl.ACL, *structs.ACLToken, error) {
 	// Fast-path if ACLs are disabled
-	if !c.config.ACLEnabled {
+	if !c.GetConfig().ACLEnabled {
```

Comment:
This is the first pattern you'll notice: config access moved to always use a getter that properly locks.

The risk here is if it is used in one of the few places that manually manages config locking you can deadlock. Tests caught the couple times that happened to me immediately, so I don't think it's a big risk.